### PR TITLE
Feature/mycroft deepspeech stt

### PR DESCRIPTION
**Pull request by LearnedVector**
## Description
Added a class to the STT factory to allow switching to deepspeech.

## How to test
deepspeech is currenty running on the test cluster so you would need to switch to api-test and change stt module to mycroft_deepspeech until the next tartarus release - 3/27/2018

```json
{  
   "stt":{  
      "module":"mycroft_deepspeech"
   },
   "server":{  
      "url":"https://api-test.mycroft.ai",
      "version":"v1",
      "update":true,
      "metrics":false
   }
}
```

## Contributor license agreement signed?
CLA [x] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/stt/__init__.py [hunk 1]
```diff
@@ -134,10 +135,29 @@ def execute(self, audio, language=None):
...
+        self.api = STTApi("deepspeech")
+
+    def execute(self, audio, language=None):
+        language = language or self.lang
+        if not language.startswith("en"):
+            raise ValueError("Deepspeech is currently english only")
+        try:
+            response = self.api.stt(audio.get_wav_data(), self.lang, 1)
+            return response
+        except exceptions.RequestException as e:
```
Comment (forslund) [inline mycroft/stt/__init__.py]: I think this needs to re-raise the error so the speech client can see if there's a 401 error and should trigger pairing

**Comment by forslund** [comment]
Looks good, worked when paired, triggered pairing when unpaired as expected.

**Comment by kjcole** [comment]
Coming to this party a bit late. 

Should I hear a change after following the steps above? The Mark 1 didn't complain, but it sounds the same it always has after adding the stanza to /home/mycroft/.mycroft/mycroft.conf

**Comment by MatthewScholefield** [comment]
@kjcole Deepspeech is STT so it only affects how Mycroft hears your voice. You should be able to verify you are on Deepspeech if you speak a sentence like `tell me about abraham lincoln` and the STT transcription contains some jibberish like `tell me about aebruhamlinkon`.

**Comment by kjcole** [comment]
Oops. Need caffeine. The whole "Darth Vader" thing led me down the Dark path to thinking it was TTS not STT.  That said, the logs don't show nonsense. mycroft-messagebus.log and mycroft-skills.log show "tell me about abraham lincoln". and it proceeded to tell me about... ;-)

**Comment by MatthewScholefield** [comment]
@kjcole Yeah, I can see how that could be confusing. As for the logs, you should most definitely see a change in accuracy at this stage. If you don't I'd look in the logs to see if there's a possibility that the client failed to load or something along those lines.

**Comment by kjcole** [comment]
All I see is:

```
9:58:32.079 - mycroft.client.speech.mic:listen:487 - DEBUG - Waiting for wake word...
19:58:32.904 - requests.packages.urllib3.connectionpool - DEBUG - https://api.mycroft.ai:443 "POST /v1/deepspeech?limit=1&lang=en-US HTTP/1.1" 200 36
19:58:32.918 - mycroft.client.speech.listener:transcribe:163 - DEBUG - STT: but tell me about abraham lincoln
19:58:32.921 - mycroft.client.speech.main:handle_utterance:61 - INFO - Utterance: ['but tell me about abraham lincoln']
```

**Comment by forslund** [comment]
The endpoint looks correct for deepspeech so should be good

**Pull request merged**